File: python/src/leetcode/container_with_most_area.py
def max_area(height: list[int]) -> int:
    left = 0
    right = len(height) - 1

    max_area = min(height[left], height[right]) * (right - left)

    while left < right:
        if height[left] < height[right]:
            left += 1
        else:
            right -= 1

        max_area = max(max_area, min(
            height[left], height[right]) * (right - left))

    return max_area

File: java/src/main/java/bparolini/dsa/neetcode/ContainerWithMostArea.java
package bparolini.dsa.neetcode;

public class ContainerWithMostArea {
  public int maxArea(int[] height) {
    int left = 0;
    int right = height.length - 1;

    int maxArea = Math.min(height[left], height[left]) * (right - left);

    while (left < right) {
      if (height[left] < height[right]) {
        left++;
      } else {
        right--;
      }

      maxArea = Math.max(maxArea,
                         Math.min(height[left], height[left]) * (right - left));
    }

    return maxArea;
  }
}
